### ISSUE TITLE
Modernize `Menu` - Base on `Bar` and address litany of issues

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2975]
Related to: 
- #2926
- #3073
- #3852 

Todo: 

- [x] #2539
- [x] Update constructors to add `Command`s and `AddKeyBinding`s instead of using `shortcut` and `action`. 
- [x] Update `ProcessKey/OnKeyPressed` handlers
- [x] Do not allow HotKeys to invoke a menu item if it is not visible. Shotcuts yes, hotkeys no. https://github.com/gui-cs/Terminal.Gui/discussions/1940#discussioncomment-7724400
- [x] `Shortcut` should be `Key` based, not `KeyCode` - 
- [x] `Hotkey` should be `Key` based, not `Rune`
- [x] #4008 
- [x] #4011
- [ ] Update `Editor` Scenario to use `MenuBarv2`
- [ ] Update `Notepad` Scenario to use `MenuBarv2`
- [ ] Mark old `Menu` classes as obsolete. 
  - [ ] Rename to `LegacyMenu` etc... 
  - [ ] Rename `Menuv2` etc... to `Menu`
- [ ] Update `MenuBar` Scenario (or nuke)
- [ ] Update `DynamicMenuBar` Scenario (or nuke or replace)
- [ ] Delete old `Menu` classes

Others:

- [ ] #2404
- [ ] #2108

_Originally posted by @tig in https://github.com/gui-cs/Terminal.Gui/discussions/2974#discussioncomment-7536482_

**Comment by tig** [comment]
Progress on this was made in #3073. Specifically:

- I developed `Shortcut` and `Bar` sufficiently that I'm confident they can be used as a basis.
- Formatting/Layout in `Shortcut` and `Bar` support the current look & feel of both `MenuBar` and `Menu`
- I introduced two temporary classes: `Menuv2` and `MenuBarv2` that are a starting point.

**Comment by tznind** [comment]
If possible can you try to retain ability to modify a existing instance after construction.

This is important for designers [Menu Operations](https://github.com/gui-cs/TerminalGuiDesigner/tree/v2/src/Operations/MenuOperations)

**Comment by tig** [comment]
> If possible can you try to retain ability to modify a existing instance after construction.
> 
> This is important for designers [Menu Operations](https://github.com/gui-cs/TerminalGuiDesigner/tree/v2/src/Operations/MenuOperations)

Yes. In fact, it should be far easier and better now. It's all just subviews of views now. No more internal lists of objects that may or may not be View subclasses with custom accessor APIs, etc... 

Just either retain references to menu items or use `SubViews` to navigate the heirarchy.

I have some known TODOs to make some key binding things get updated, but I fully intend to make this work before this Issue is closed.

**Comment by tig** [comment]
To prove this works, my plan is to refactor UICatalog's Theme menu to actually be dynamically generated when shown (right now it is  recreated whenever the config changes).
